### ISSUE TITLE
🐛 Bug Report: Possibly wrong import path in plugin dist files when using both css and svg

### ISSUE THREAD

**Issue by jslott2sigma** [backstage/backstage#25954]
### 📜 Description

We've noticed that the import path for an imported svg file in the `dist` output of `yarn build` is relative to the source code tree and not the dist tree when importing both a `css` and `svg`. As a result the svg is not found when using the plugin in a front-end app.

### 👍 Expected behavior

If we define (and export) the following React component from a plugin that uses an `svg` file within the same directory:
```
import React from 'react';
import logo from './logo.svg';

export function Example(): JSX.Element {
    return (
        <div>
            <img src={logo} />
        </div>
    );
}
```
upon `yarn build`, the file looks like:
```
import React from 'react';
import logo from './logo.svg';

function Example() {
  return /* @__PURE__ */ React.createElement("div", null, /* @__PURE__ */ React.createElement("img", { src: logo }));
}

export { Example };
//# sourceMappingURL=ExamplePage.esm.js.map
```
**(note the path of the relative import for `logo`)**

### 👎 Actual Behavior with Screenshots

Instead, if the React component includes both a `css` and `svg` like such:
```
import React from 'react';
import logo from './logo.svg';
import styles from './Example.css';

export function Example(): JSX.Element {
    return (
        <div className={styles.NoMargin}>
            <img src={logo} />
        </div>
    );
}
```
the compiled file in the `dist` directory is instead:
```
import React from 'react';
import logo from '../../plugins/example/src/components/ExamplePage/logo.svg';
import css_248z from './Example.css.esm.js';

function Example() {
  return /* @__PURE__ */ React.createElement("div", { className: css_248z.NoMargin }, /* @__PURE__ */ React.createElement("img", { src: logo }));
}

export { Example };
//# sourceMappingURL=ExamplePage.esm.js.map
```
**(note the relative directory for the `logo` import).**

### 👟 Reproduction steps

1. In a plugin, create a simple React component as above, export the component from the plugin
2. Run `yarn build`
3. Observe the import path in the compiled file in the `dist` directory.

### 📃 Provide the context for the Bug.

This was against 1.28.4. We do not believe this happens in 1.26.4. I know this seems like an unusual bug, so many thanks for your help!

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

**Comment by Rugvip** [comment]
Thank you for reporting! 👍 

This is likely triggered by #23417, but I don't think it's the root cause. I poked around a bit, assuming this is a bug in our [file forwarding plugin](https://github.com/backstage/backstage/blob/5bf2c128fb44f87aded075a5a1c6efb76a7e4a27/packages/cli/src/lib/builder/plugins.ts#L40), but it actually seems to behave correctly.

This issue is fixed if the `inject` option of the `postcss` plugin is disabled, or if it's configured to avoid the absolute file import that `style-inject` usually does, e.g. like this:

```ts
  postcss({
    inject: cssVariableName => `
import styleInject from 'style-inject';

styleInject(${cssVariableName});
`,
  })
```

The problem with that is that with the above code we leave `style-inject` to be imported when bundling instead, and since it won't be a dependency of the published package, that might either break because it's missing, or because the wrong version is used. It's a possible fix for this issue though, since `style-inject` hasn't gotten a new version in 7 years.

In the end I'm not quite sure where the issue is tbh. It seems to be somewhere in the output processing where the `./logo.svg` import that should have correctly been marked as external gets rewritten to the incorrect path that seems to be relative to the package path. That in itself is strange because we don't use the exact package path anywhere in config either, so this resolution could be happening towards `process.cwd()` under the hood.

**Comment by jslott2sigma** [comment]
@Rugvip Would it be possible to consider another `rollup` plugin that handles the `css`? Using this seems to fix the problem I was seeing:
```
import styles from 'rollup-plugin-styles';
...
plugins[
   ...
   styles({ mode: 'extract' }),
   ...
]
```
I'm not an expert however on `postcss` nor `style-inject`, however it seems the `styles` rollup plugin is much more current than the `postcss` plugin.

Thanks!

**Comment by Rugvip** [comment]
@jslott2sigma yep that sounds good! If you're up for opening a PR for that I can verify that the build output looks alright

**Comment by jslott2sigma** [comment]
@Rugvip Yup, here you go! https://github.com/backstage/backstage/pull/26111

**Comment by Rugvip** [comment]
Happen to stumble upon a fix for this in #26401
